core/src/main/java/org/elasticsearch/index/query/BoolQueryParser.java

```diff
@@ -78,12 +78,12 @@ public BoolQueryBuilder fromXContent(QueryParseContext parseContext) throws IOEx
                     shouldClauses.add(query);
                     break;
                 case "filter":
-                    query = parseContext.parseInnerFilterToQueryBuilder();
+                    query = parseContext.parseInnerQueryBuilder();
```

Comment:
cool, we just had to do it twice while queries were going to be moved over, now we do it in a single place. Thanks for clarifying.